function.php

```diff
@@ -0,0 +1,107 @@
+<?php
+require_once 'db.php';
+
+/**
+ * getData function gets data from the DB
+ * @param $db
+ * @return array
+ */
+function getData(PDO $db, string $request):array {
+
+    $db->setAttribute(
+        PDO::ATTR_DEFAULT_FETCH_MODE,
+        PDO::FETCH_ASSOC);
+
+    $sql = $db->prepare($request);
+    $sql->execute();
+    $groceryItems = $sql->fetchAll();
+
+    return $groceryItems;
+}
+
+/**
+ * generateRequest function takes $filter as param and returns a query to retrieve data from DB. It was created for unit testing, it took some part from getData function
+ * @param string $filter
+ * @return string
+ */
+function generateRequest(string $filter = 'all'):string {
+    $requestFilters = '';
+    if ($filter !== 'all') {
+        $requestFilters = ' WHERE `category` = "' . $filter . '";';
+    }
+
+    $request = 'SELECT `item`, `category`, `price`, `remaining` FROM `grocery_item`'.$requestFilters;
+    return $request;
+}
+/**
+ * processData function takes return value of getData function and returns all data from DB in HTML unit
+ * @param $groceryItems
+ * @return string
+ */
+function processData(array $groceryItems):string {
+    $itemRow= '';
+    foreach($groceryItems as $item) {
+        $itemRow .= '<div class="item-row"><ul><li><span>Item</span>' . $item["item"] . '</li><li class="' .$item["category"] . '-row">' . $item['category'] . '</li><li><span>Price(pense)</span>' . $item['price'] . '</li><li><span>Remaining(%)</span>' . $item['remaining'] . '</li></ul></div>';
```

Comment:
Storing the price in pence is one thing but for presentation shouldn't it be in pounds?

Reply:
I want to but I don't know how to add full stop between numbers